lib/dtutils.lua

```diff
@@ -0,0 +1,457 @@
+--[[
+
+    dtutils.lua - common darktable lua functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils = {}
+
+dtutils.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "See_Also", "License"},
+  Name = [[dtutils - A Darktable lua utilities library]],
```

Comment:
I am thinking of non-manpage generation here
* name : just the name
* synopsis : one line description 
* usage : mainly arguments, return values and their description
* example : actual code